src/mpyl/artifacts/build_artifacts.py

```diff
@@ -1,85 +1,62 @@
-        self.cache_folder = self.artifact_repo.root_dir / CACHE_FOLDER_NAME
-
-    def get_build_artifacts_paths(self) -> list[Path]:
-        found_projects: list[Path] = [
-            Path(
-                load_project(
-                    self.codebase_repo.root_dir, Path(project_path), strict=False
-                ).target_path
-            )
-            for project_path in self.codebase_repo.find_projects()
-        ]
-
-        return [path for path in found_projects if path.exists()]
+        self.artifact_repo_config = artifact_repo_config
 
     def pull(self, branch: str) -> None:
-        if self.artifact_repo.does_branch_exist(branch_name=branch, remote=True):
-            self.logger.info(f"Fetching branch '{branch}' from remote")
-            self.artifact_repo.fetch_branch(branch_name=branch)
-            self.artifact_repo.checkout_branch(branch_name=branch)
-
-            for artifact_path in os.listdir(self.cache_folder):
-                shutil.copytree(
-                    src=self.cache_folder / artifact_path,
-                    dst=artifact_path,
-                    dirs_exist_ok=True,
-                )
-        else:
-            self.logger.info(f"Branch '{branch}' has no remote to pull from")
-
-    def push(self, branch: str) -> None:
-        if self.artifact_repo.does_branch_exist(branch_name=branch):
-            if self.artifact_repo.get_branch != branch:
-                self.artifact_repo.checkout_branch(branch_name=branch)
-        else:
-            if self.artifact_repo.does_branch_exist(branch_name=branch, remote=True):
-                self.logger.info(f"Fetching branch '{branch}' from remote")
-                self.artifact_repo.fetch_branch(branch_name=branch)
-                self.artifact_repo.checkout_branch(branch_name=branch)
-            else:
-                self.logger.info(f"Creating new branch '{branch}'")
-                self.artifact_repo.create_branch(branch_name=branch)
-
-        shutil.rmtree(self.cache_folder, ignore_errors=True)
-
-        for artifact_path in self.get_build_artifacts_paths():
-            shutil.copytree(
-                src=artifact_path,
-                dst=self.cache_folder / artifact_path,
-                dirs_exist_ok=True,
-            )
-
-        if self.artifact_repo.has_changes():
-            self.logger.info("Committing and pushing all artifacts")
-            self.artifact_repo.stage(CACHE_FOLDER_NAME)
-            self.artifact_repo.commit(f"Add artifacts for {branch}")
-            self.artifact_repo.push()
+        with TemporaryDirectory() as tmp_repo_dir:
+            with Repository.from_clone(
+                config=self.artifact_repo_config, repo_path=Path(tmp_repo_dir)
+            ) as artifact_repo:
+                if artifact_repo.does_branch_exist(branch_name=branch):
+                    self.logger.info(f"Fetching branch '{branch}' from remote")
+                    artifact_repo.checkout_branch(branch_name=branch)
```

Comment:
We have to copy the files out of the repo here as well if you want to use `with TemporaryDirectory`

Reply:
Yes, we only need to take the caching scenario into account, deploy artifacts are never popped.